src/components/tl/ucp/tl_ucp_context.c

```diff
@@ -579,10 +459,24 @@ ucc_status_t ucc_tl_ucp_get_context_attr(const ucc_base_context_t *context,
         }
     }
     if (attr->attr.mask & UCC_CONTEXT_ATTR_FIELD_CTX_ADDR_LEN) {
-        attr->attr.ctx_addr_len = ctx->ucp_addrlen;
+        packed_length = ctx->ucp_addrlen + ctx->n_rinfo_segs * (sizeof(size_t) * 3);
+        for (int i = 0; i < ctx->n_rinfo_segs; i++) {
+            packed_length += ctx->remote_info[i].packed_key_len;
+        }
+        attr->attr.ctx_addr_len = packed_length; 
     }
     if (attr->attr.mask & UCC_CONTEXT_ATTR_FIELD_CTX_ADDR) {
         memcpy(attr->attr.ctx_addr, ctx->worker_address, ctx->ucp_addrlen);
+        if (NULL != ctx->remote_info) {
+            ucc_status = ucc_tl_ucp_ctx_remote_pack_data(ctx, &packed_data, &packed_length);
```

Comment:
isn't it a mem leak here? you allocate packed_data inside ucc_tl_ucp_ctx_remote_pack_data call , do memcpy but never free the packed_data array. Do you even need it? The core level will first call this get_attr routine with just UCC_CONTEXT_ATTR_FIELD_CTX_ADDR_LEN  bit and query the required space, then it will provide you a buffer that you can directly use.

Reply:
You are correct. I've removed leak as you have suggested.